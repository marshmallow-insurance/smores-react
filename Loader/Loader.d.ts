import { FC } from 'react';
import { Color } from '../theme';
export type LoaderProps = {
    height: string;
    color?: Color;
};
export declare const Loader: FC<LoaderProps>;
