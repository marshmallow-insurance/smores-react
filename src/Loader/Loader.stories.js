import React from 'react'
import { Loader } from '../Loader'

export default {
  title: 'Loader',
  component: Loader,
}

const Template = (args) => <Loader {...args} />

export const Default = Template.bind({})

Default.args = {
  height: 32,
}

export const BigAndBlue = Template.bind({})

BigAndBlue.args = {
  height: 54,
  color: 'blue6',
}
