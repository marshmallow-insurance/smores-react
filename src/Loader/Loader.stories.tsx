import React from 'react'
import { Loader, LoaderProps } from './Loader'

export default {
  title: 'Loader',
  component: Loader,
}

const Template = (props: LoaderProps) => <Loader {...props} />

export const Default = Template.bind({})

Default.args = {
  height: 32,
}

export const BigLoader = Template.bind({})

BigLoader.args = {
  height: 54,
  color: 'liquorice',
}
