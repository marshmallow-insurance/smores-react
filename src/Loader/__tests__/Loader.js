import React from 'react'
import { render } from '@testing-library/react'
import 'jest-styled-components'

import { Loader } from '../Loader'

test('renders', () => {
  const { container } = render(<Loader color="marshmallowPink" />)
  expect(container.firstChild).toMatchSnapshot()
})
