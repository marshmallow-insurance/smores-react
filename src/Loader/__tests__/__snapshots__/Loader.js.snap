// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.c0 path {
  fill: #FF88C8;
  stroke: none;
  fill-rule: evenodd;
}

.c1 {
  -webkit-animation: Marshmallow1_t 1.6s linear infinite both, Marshmallow1_d 1.6s linear infinite both;
  animation: Marshmallow1_t 1.6s linear infinite both, Marshmallow1_d 1.6s linear infinite both;
}

.c2 {
  -webkit-animation: Marshmallow2_t 1.6s linear infinite both, Marshmallow2_d 1.6s linear infinite both;
  animation: Marshmallow2_t 1.6s linear infinite both, Marshmallow2_d 1.6s linear infinite both;
}

.c3 {
  -webkit-animation: Marshmallow3_t 1.6s linear infinite both, Marshmallow3_d 1.6s linear infinite both;
  animation: Marshmallow3_t 1.6s linear infinite both, Marshmallow3_d 1.6s linear infinite both;
}

<svg
  class="c0"
  color="marshmallowPink"
  viewBox="0 0 60 32"
>
  <g
    transform="translate(30,18.5455) translate(-30,-9)"
  >
    <path
      class="c1"
      d="M73.999,81.9767L73.6592,70.6087C73.6352,69.8071,73.1892,69.0806,72.4916,68.707C71.1747,68.0014,69.0105,67.648,66.0001,67.6467C62.9894,67.648,60.8254,68.0014,59.5084,68.707C58.8109,69.0806,58.3648,69.8071,58.3408,70.6087L58.001,81.9767C57.983,82.5754,58.2138,83.1538,58.6366,83.5705C60.0408,84.9546,62.4906,85.6467,65.986,85.6467C65.9907,85.6467,65.9953,85.6467,66.0001,85.6467C66.0047,85.6467,66.0094,85.6467,66.014,85.6467C69.5095,85.6467,71.9593,84.9546,73.3634,83.5705C73.7863,83.1538,74.0169,82.5754,73.999,81.9767"
      id="Marshmallow1"
      transform="translate(-58,-67.6467)"
    />
    <path
      class="c2"
      d="M73.999,81.9767L73.6592,70.6087C73.6352,69.8071,73.1892,69.0806,72.4916,68.707C71.1747,68.0014,69.0105,67.648,66.0001,67.6467C62.9894,67.648,60.8254,68.0014,59.5084,68.707C58.8109,69.0806,58.3648,69.8071,58.3408,70.6087L58.001,81.9767C57.983,82.5754,58.2138,83.1538,58.6366,83.5705C60.0408,84.9546,62.4906,85.6467,65.986,85.6467C65.9907,85.6467,65.9953,85.6467,66.0001,85.6467C66.0047,85.6467,66.0094,85.6467,66.014,85.6467C69.5095,85.6467,71.9593,84.9546,73.3634,83.5705C73.7863,83.1538,74.0169,82.5754,73.999,81.9767"
      id="Marshmallow2"
      transform="translate(-36,-67.6467)"
    />
    <path
      class="c3"
      d="M73.999,81.9767L73.6592,70.6087C73.6352,69.8071,73.1892,69.0806,72.4916,68.707C71.1747,68.0014,69.0105,67.648,66.0001,67.6467C62.9894,67.648,60.8254,68.0014,59.5084,68.707C58.8109,69.0806,58.3648,69.8071,58.3408,70.6087L58.001,81.9767C57.983,82.5754,58.2138,83.1538,58.6366,83.5705C60.0408,84.9546,62.4906,85.6467,65.986,85.6467C65.9907,85.6467,65.9953,85.6467,66.0001,85.6467C66.0047,85.6467,66.0094,85.6467,66.014,85.6467C69.5095,85.6467,71.9593,84.9546,73.3634,83.5705C73.7863,83.1538,74.0169,82.5754,73.999,81.9767"
      id="Marshmallow3"
      transform="translate(-14,-67.6467)"
    />
  </g>
</svg>
`;
